### PR TITLE
Add many, many missing types to PostgresDataType

### PR DESCRIPTION
**Pull request by gwynne**
_**IMPORTANT**: This PR only deals with defining the well-known OIDs for Postgres built-in types. Except for adding more array/element type mappings, it does **NOT** add any additional support for any of these types, not even basic encoding and decoding! That support will be a separate PR._

Including but not limited to:
- Several geometry types (`lseg`, `path`, `box`, `polygon`, `line`)
- The `reg*` types (`regclass`, `regtype`, etc.)
- Missing network address types (`macaddr8`, `inet`, `cidr`)
- Array types for both existing and other new types
- `bit` and `bit varying`
- The pseudo-types (`cstring`, `internal`, `any`, `anycompatible`, etc.)
- The text search types (`tsvector`, `tsquery`)
- Missing range types (`numrange`, `tsrange`, `tstzrange`, `daterange`)
- `jsonpath`
- The multirange types (`int4multirange`, `nummultirange`, etc.)
- Some special-purpose types (`tid`, `xid`, `cid`, `xid8`, `pg_lsn`, `trigger`, etc.)

Also adds full two-way mappings for every known array type (including existing ones that were missing the mapping, like `money`).

Requires minor version bump (adds new public API and deprecates `PostgresDataType.pgNodeTree`)

### CODE REVIEW

**Review by fabianfett — COMMENTED** [review]
Can we add a comment where the source of truth is?

**Comment by gwynne** [comment]
> Can we add a comment where the source of truth is?

The source of truth I used is the actual `pg_type` table in a fresh database as it appears in my installation of Postgres 15, with a bunch of "**_really_** internal" types removed by eye; there's not really any one place to point for truth other than the Postgres docs.

File: Sources/PostgresNIO/Data/PostgresDataType.swift [hunk 1]
```diff
@@ -49,18 +51,75 @@ public struct PostgresDataType: RawRepresentable, Sendable, Hashable, CustomStri
...
+    public static let pgDDLCommand = PostgresDataType(32)
     /// `114`
     public static let json = PostgresDataType(114)
+    /// `142`
+    public static let xml = PostgresDataType(142)
+    /// `143`
+    public static let xmlArray = PostgresDataType(143)
     /// `194` pg_node_tree
+    @available(*, deprecated, message: "This is internal to Postgres and should not be used.")
     public static let pgNodeTree = PostgresDataType(194)
```
[the remark this answers is not in the document]
Comment (gwynne) [reply Sources/PostgresNIO/Data/PostgresDataType.swift:72]: It's historically `public`, making it private would be source-breaking (semver-major)

File: Sources/PostgresNIO/Data/PostgresDataType.swift [hunk 2]
```diff
@@ -49,18 +51,75 @@ public struct PostgresDataType: RawRepresentable, Sendable, Hashable, CustomStri
...
+    public static let circleArray = PostgresDataType(719)
+    /// `774`
+    public static let macaddr8 = PostgresDataType(774)
+    /// `775`
+    public static let macaddr8Aray = PostgresDataType(775)
     /// `790`
     public static let money = PostgresDataType(790)
+    /// `791`
+    @available(*, deprecated, renamed: "moneyArray")
+    public static let _money = PostgresDataType(791)
```
[the remark this answers is not in the document]
Comment (gwynne) [reply Sources/PostgresNIO/Data/PostgresDataType.swift:117]: It's historically `public`, making it private would be source-breaking (semver-major)

**Review by fabianfett — APPROVED** [review]

**Pull request merged**